apps/mainsite/management/commands/list_esco_issuers.py

```diff
@@ -0,0 +1,56 @@
+from json import loads
+from django.core.management.base import BaseCommand
+from issuer.models import BadgeClass
+import os
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        # Structure: {issuer_name: {badge_id: [esco_ids]}}
+        issuer_data = {}
+        
+        badgeclasses = BadgeClass.objects.all()
+        
+        file_path = os.path.join(os.getcwd(), 'esco_issuers.txt')
+        try:
+            for badgeclass in badgeclasses:
+                extensions = badgeclass.get_extensions_manager()
+                competency_extension = extensions.filter(name='extensions:CompetencyExtension').first()
+                
+                if competency_extension is not None:
+                    competency_json = competency_extension.original_json
+                    competency_dict = loads(competency_json)
+                    
+                    esco_ids = []
+                    for item in competency_dict:
+                        escoID = item.get('escoID')
+                        if escoID is not None and escoID != '':
+                            esco_ids.append(escoID)
+                    
+                    if esco_ids:  
+                        issuer_name = badgeclass.issuer.name
+                        badge_id = badgeclass.entity_id
+                        
+                        if issuer_name not in issuer_data:
+                            issuer_data[issuer_name] = {}
+                        
+                        issuer_data[issuer_name][badge_id] = esco_ids
+            
+            with open(file_path, 'w') as f:
+                for issuer_name, badges_data in issuer_data.items():
+                    f.write(f"\nIssuer: {issuer_name}\n")
```

Comment:
Sadly there wasn't a description in the ticket, so I don't know what this script is supposed to be used for. But I have a few questions:
- Doesn't this imply that the file starts with a blank line? Why would we want that?
- Why do we not use a more machine-readable format, like CSV?

Reply:
I see, i added the description to the ticket now. Since the script will only be used once and the output will not be used anywhere else i didnt bother with the format too much